src/gradualizer_lib.erl

```diff
@@ -168,6 +168,12 @@ pick_value(?type(record, [{atom, _, RecordName} | Tys]), TEnv) ->
         || ?type(field_type, [{atom, _, FieldName}, Ty]) <- Tys
     ],
     {record, erl_anno:new(0), RecordName, MFields};
+pick_value(?type(map, Assocs), TEnv) ->
+    NewAssocs = [ {AssocTag, Anno, pick_value(KTy, TEnv), pick_value(VTy, TEnv)}
+                  || {type, Anno, AssocTag, [KTy, VTy]} <- Assocs ],
+    {map, erl_anno:new(0), NewAssocs};
+pick_value(?type(boolean), _TEnv) ->
+    {atom, erl_anno:new(0), false};
```

Comment:
Since boolean is just `false | true`, I guess in the future we should expand this to a union and then be able to pick a value from a union?

Reply:
Hmm, I'm not sure of the extent of it, but some types are passed around denormalised. For example, see https://github.com/josefs/Gradualizer/blob/0b8e6a8fe1100b8e11cd6f822c9093ee76959eaa/src/typechecker.erl#L379 - I think some denormalised types reaching `pick_value` might come from here.